src/extensions/app-extension-popup-state.ts

```diff
@@ -53,8 +61,24 @@ const inactiveState = new ExtensionInactiveState();
 
 const stateAtom = atom<AppPopupExtensionPossibleStates>(inactiveState);
 
+/**
+ * TODO: Make this to work also with widget extensions
+ */
 export const useAppExtensionPopup = () => {
   const [state, setState] = useAtom(stateAtom);
+  const { iframes } = usePopupFrameReference();
+
+  useEffect(() => {
+    if (state.active) {
+      iframes.entries().forEach(([iframe, { loaded }]) => {
+        if (loaded) {
+          const origin = new URL(state.src).origin;
+
+          postToExtension(DashboardEventFactory.createFormEvent(state.formState!), iframe, origin);
+        }
+      });
+    }
+  }, [state.active, iframes]);
```

Comment:
check if this works:
1. app is active:
2. wait 500ms
3. app receives formUpdate from product page

is this sending event?